File: JavaScript/c01_intro/e03-check-palindrome.js
function checkPalidrome(inputString) {
    return inputString === inputString.split('').reverse().join('');
}

const testCases = [
    'aabaa',
    'abac',
    'a',
    'az',
    'abacaba',
    'z',
    'aaabaaaa',
    'zzzazzazz',
    'hlbeeykoqqqqokyeeblh',
    'hlbeeykoqqqokyeeblh'
];

testCases.forEach((input) => {
    const output = checkPalidrome(input);
    console.log(`Input: ${input},`.padEnd(30, ' ') + `Output: ${output}`);
});

File: Python/c01_intro/e03_check_palindrome.py
def check_palindrome(input_string):
    return input_string == input_string[::-1]


def test():
    test_cases = [
        "aabaa",
        "abac",
        "a",
        "az",
        "abacaba",
        "z",
        "aaabaaaa",
        "zzzazzazz",
        "hlbeeykoqqqqokyeeblh",
        "hlbeeykoqqqokyeeblh"
    ]

    for test_case in test_cases:
        input_str = f"Input: {test_case}"
        print(input_str.ljust(30) + "Output:", check_palindrome(test_case))


def main():
    test()


if __name__ == "__main__":
    main()
